fix(ui): focus manager fixes and improvements

milesap opened the pull request:
Fixes an issue where the focus management waiting dialog remains on the page
in standalone maps and can crash the browser.

This fix also improves the focus manager with better event
timing and detection.

Closes #1111

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/fgpv-vpgf/fgpv-vpgf/1167)
<!-- Reviewable:end -->

AleksueiR commented:
Found some problems:

#### Applies to all samples pages
- the "keyboard focus" dialog will not be shown after any mouse movements when you tab to the viewer afterwards
- pressing "Enter" on the "keyboard focus" dialog immediately triggers identify and opens a details panel
- pressing "Shift + Enter" on the "keyboard focus" dialog mistakenly activates the viewer
- mouse clicking on any button immediately after the page loads brings up the "keyboard focus" dialog
- pressing "Tab" or "Shift + Tab" on the "keyboard focus" dialog has a considerable delay before moving focus to the next or previous focusable element outside the viewer; if "Tab" or "Shift + Tab" is pressed again during this delay, it is reset back

#### index-one
- pressing "Shift + Tab" when the "keyboard focus" dialog is open, hides it and immediately brings it back up

#### index-many
- tabbing through one viewer instance will move the focus to the next on on the page when reaching the last focusable element; focus traverses all viewer instances on the page and returns to the initial one

---

Review status: 0 of 1 files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

*[src/app/layout/focus.service.js, line 41 at r1](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1167#-KRmobiWWZal-KsrnaNv:-KRmobiWWZal-KsrnaNw:be5swvi) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/e5109881a5a8c0a18573643098a1d9ac1388cbba/src/app/layout/focus.service.js#L41)):*
> ```JavaScript
> 
>         // detect if there has been a mousedown, if so disable focus management entirely
>         $(document).on('mousedown', () => dlg.hide().then(_setStatus));
> ```

Same deal with underscores as in the previous PR - they should be used only for private variables/functions when they are exposed to other code.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1167#-:-KRml9SSbI38VMjdlcS1:b-nbvw3v)*
<!-- Sent from Reviewable.io -->

milesap commented:
- fixed
- separate issue as discussed
- fixed
- fixed
- fixed
- tested and working for me (no "keyboard focus" dialog is shown, focus management is automatic)
- tested and working for me (please confirm if this is still an issue)

---

Review status: 0 of 1 files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

*[src/app/layout/focus.service.js, line 41 at r1](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1167#-KRmobiWWZal-KsrnaNv:-KRn6xBXEgTG7SqY-pbz:b2sba8y) ([raw file](https://github.com/fgpv-vpgf/fgpv-vpgf/blob/e5109881a5a8c0a18573643098a1d9ac1388cbba/src/app/layout/focus.service.js#L41)):*
<details><summary><i>Previously, AleksueiR (Aleksuei Riabtsev) wrote…</i></summary>
> Same deal with underscores as in the previous PR - they should be used only for private variables/functions when they are exposed to other code.
</details>
Ok removed `_`

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1167#-:-KRn-b0iehcTtlxlThhq:b-3dcjxf)*
<!-- Sent from Reviewable.io -->

AleksueiR commented:
- :ballot_box_with_check: 
- see this https://github.com/fgpv-vpgf/fgpv-vpgf/issues/1191
- :ballot_box_with_check: 
- :ballot_box_with_check: 
- :ballot_box_with_check: 

- :ballot_box_with_check: 

- see this https://github.com/fgpv-vpgf/fgpv-vpgf/issues/834

---

Review status: 0 of 1 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1167#-:-KRnCMQENQRVZcoaQYRo:b-x63rbh)*
<!-- Sent from Reviewable.io -->

milesap commented:
Added a "work-around" for the last point until #834 is resolved

---

Review status: 0 of 1 files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1167#-:-KRnPrwNJC0Id3N-1ogf:b8i8q1d)*
<!-- Sent from Reviewable.io -->

AleksueiR commented:
See issue https://github.com/fgpv-vpgf/fgpv-vpgf/issues/1192

---

Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/fgpv-vpgf/fgpv-vpgf/1167#-:-KRo10j9nAYQuwrJkrJF:b-xuwc1v)*
<!-- Sent from Reviewable.io -->
